doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,235 @@
+ophthalmologists to take pictures of the back of a patient's eye [1]_.
+When performing OCT,
+dust may stick to the reference mirror of the equipment, causing dark spots to
+appear on the images. The problem is that these dark spots cover areas of
+in-vivo tissue, hence hiding data of interest. Our goal here is to restore
+(reconstruct) the hidden areas based on the pixels near their boundaries.
+
+This tutorial is adapted from an application shared by Jules Scholler [2]_.
+The images were acquired by Viacheslav Mazlin.
+
+.. [1] David Turbert, reviewed by Ninel Z Gregori, MD (2023)
+       `What Is Optical Coherence Tomography?
+       <https://www.aao.org/eye-health/treatments/what-is-optical-coherence-tomography>`_,
+       American Academy of Ophthalmology.
+.. [2] Jules Scholler (2019) "Image denoising using inpainting"
+       https://www.jscholler.com/2019-02-28-remove-dots/
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import plotly.io
+import plotly.express as px
+import plotly.graph_objects as go
+from scipy import sparse
+
+from skimage import (
+    filters, measure, morphology, restoration
+)
+from skimage.data import palisades_of_vogt
+
+
+#####################################################################
+# The dataset we are using in this example is an image sequence (a movie!) of
+# human in-vivo tissue. Specifically, it shows the *palisades of Vogt* of a
+# given cornea sample.
+
+#####################################################################
+# Load image data
+# ===============
+
+image_seq = palisades_of_vogt()
+
+print(f'number of dimensions: {image_seq.ndim}')
+print(f'shape: {image_seq.shape}')
+print(f'dtype: {image_seq.dtype}')
+
+#####################################################################
+# The dataset is an image stack with 60 frames (time points) and 2 spatial
+# dimensions. We can
+# visualize it by taking advantage of the ``animation_frame`` parameter in
+# Plotly's ``imshow`` function. As a side note, when the
+# ``binary_string`` parameter is set to ``True``, the image is
+# represented as grayscale.
+
+fig = px.imshow(
+    image_seq,
+    animation_frame=0,
+    height=500,
+    width=500,
+    binary_string=True,
+    labels={'animation_frame': 'time point'},
+    title='Sample of in-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Average over time
+# =================
+# First, we want to detect those dark spots where the data are lost. In
+# technical terms, we want to *segment* the dark spots (for
+# all frames in the sequence). Unlike the actual data (signal), the dark spots
+# do not move from one frame to the next; they are still. Therefore, we begin
+# by computing the time average of the image sequence. We shall use this
+# time-averaged image to segment the dark spots, the latter then standing out
+# with respect to the background (blurred signal).
+
+image_avg = np.mean(image_seq, axis=0)
```

Comment:
> Any sense in using `np.median` as a more robust statistical measure here?
> 
> I would also find the variance between frames interesting as part of the "getting a feel for the data part". Basically, add
> 
> ```python
> image_variance = np.var(image_seq, axis=0)
> ```
> 
> as a second image in the figure below. What do you think?
Great suggestion. Statistically median is a more reliable measure than mean. Since output from `np.median` is similar to `np.mean`. For robustness, `np.median` should be favored.

Reply:
@decorouz @ana42742 have you tested whether using the median instead of the mean yielded a different segmentation? If so, *how* different?

Reply:
The segmentation result is indeed slightly different, see below.

* Using `np.mean`:
![with_mean](https://github.com/scikit-image/scikit-image/assets/2227806/13947198-150f-46af-995e-16934de723cd)

* Using `np.median`:
![with_median](https://github.com/scikit-image/scikit-image/assets/2227806/23da1912-5bcb-42d2-be6a-6ffacbc07c11)

It's not *strikingly* better, a few spots are lost and others are found; overall it's a slight improvement (67 vs. 64 spots found, from running `ski.measure.label(mask_dilate)`). Done 032c56996e7d8.

Reply:
> @decorouz @ana42742 have you tested whether using the median instead of the mean yielded a different segmentation? If so, _how_ different?

Yes, I have tested it, my observation is similar as you mentioned. Slight improvement. I do upvote the use of `np.mean`